app/assets/javascripts/training/components/quiz.jsx

```diff
@@ -29,10 +29,6 @@ const Quiz = createReactClass({
     return this.setState({ selectedAnswerId: e.currentTarget.getAttribute('data-answer-id') });
   },
 
-  UNSAFE_componentWillReceiveProps(newProps) {
```

Comment:
Remind me what's up with this file. I'm confused about how we're using state vs. props for keeping track of the selected answer. Was this bit of copying props over to state just unnecessary?

Are there better names we could use than `setAnswer` vs `setSelectedAnswer` to make it clear what's going on?

Reply:
Okay...So through `setAnswer`, we first update the choice user has chosen from the `e.currentTarget`. This update is done through setState (ofc), which means that the state is updated successfully. Now that this part is done as and when the user wants to change the option, now the user clicks on `CHECK ANSWER` which calls verifyAnswer which in turn calls `setSelectedAnswer`. Now, what `setSelectedAnswer` does is, that it updates the redux state through the action appropriately so that the user is able to go to the next slide if the option chosen is the same as `state.currentSlide.assessment.correct_answer_id`. `this.props.selectedAnswerId` is actually nothing in the process I just elaborated and so always comes out to be `undefined`. This is why I removed it.